README.md

```diff
@@ -73,6 +76,38 @@ On Android permission is required to write in the external storage when saving t
 <application>
 ```
 
+Then you have to explicitly ask for the permission
+
+```javascript
+import { PermissionsAndroid, Platform } from "react-native";
+import CameraRoll from "@react-native-community/cameraroll";
+
+const checkAndroidPermission = async () => {
```

Comment:
If we still wanted to include some sample code (which I don't think is needed), here's how I'd restructure this code to be more succinct. It solves 2 specific problems:

1. Promise.resolve() does nothing in an async function
2. We still unnecessarily call the `saveToCameraRoll` function even if we don't have the permission

```js
import { PermissionsAndroid, Platform } from "react-native";
import CameraRoll from "@react-native-community/cameraroll";

async function hasAndroidPermission() {
  const permission = PermissionsAndroid.PERMISSIONS.WRITE_EXTERNAL_STORAGE;
  let hasPermission = await PermissionsAndroid.check(permission);
  if (hasPermission) {
    return true;
  }
  const status = await PermissionsAndroid.request(permission);
  return status === 'granted';
}

async function savePicture() {
  if (Platform.OS === "android" && !(await hasAndroidPermission())) {
    return;
  }
  CameraRoll.saveToCameraRoll(tag, [type]);
};
```

Reply:
Yes, that makes sense. Thanks. I would want to change `let` to `const` though as it serves no purpose being a `let`.